typechecker/src/type_evaluator.rs

```diff
@@ -59,6 +60,13 @@ bitflags::bitflags! {
     }
 }
 
+fn class_type_to_instance_type(class_type: PythonType) -> PythonType {
```

Comment:
whats an example python > c++ translation this function is used for? i couldn't understand the logic here 😃

Reply:
The Python type checker distinguishes between a "class type" like for example in
```
class A:
   pass

type(A) => class A
a = A()
type(a) => instance A
```
and this code converts from a class type to an instance type. This is used by the Python typechecking logic (not for the C++ translation). I think the original enderpy code had some bugs where they used class types instead of instance types, and this function converts them so the typechecking can work out.